### PR TITLE
Adds Dockerfiles for bwa, bwa-samtools, and gatk-base

### PR DESCRIPTION
**Pull request by dleehr**
Tools are used by bespin-cwl exomeseq and follow new versioning conventions. I will update the Docker Hub builds for these once merged into master.

### CODE REVIEW

File: bwa-samtools/0.7.12_1.2.1-254-6462e34/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    apt-get clean && \
+    apt-get purge && \
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*
+
+### Installing samtools/htslib/tabix/bgzip
+
+ENV VERSIONH 1.2.1-254-6462e34
+ENV NAMEH htslib
+ENV URLH "https://github.com/samtools/htslib/archive/${VERSIONH}.tar.gz"
+ENV SHA1H "6462e349d16e83db8647272e4b98d2a92992071f"
```
Comment (johnbradley) [inline bwa-samtools/0.7.12_1.2.1-254-6462e34/Dockerfile]: I'm a little confused. Are we using URLH somewhere? I assume `6462e349d16e83db8647272e4b98d2a92992071f` matches up to URLH?

Comment (dleehr) [inline bwa-samtools/0.7.12_1.2.1-254-6462e34/Dockerfile]: Ah good catch. The part that uses it was commented out in the original. So I deleted the commented-out stuff but didn't catch the unused var. https://github.com/common-workflow-language/workflows/blob/master/tools/samtools-Dockerfile#L54

File: bwa-samtools/0.7.12_1.2.1-254-6462e34/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+### Installing samtools/htslib/tabix/bgzip
+
+ENV VERSIONH 1.2.1-254-6462e34
+ENV NAMEH htslib
+ENV SHA1H "6462e349d16e83db8647272e4b98d2a92992071f"
+
+ENV VERSION 1.2-242-4d56437
+ENV NAME "samtools"
+ENV URL "https://github.com/samtools/samtools/archive/${VERSION}.tar.gz"
```
Comment (johnbradley) [inline bwa-samtools/0.7.12_1.2.1-254-6462e34/Dockerfile]: One more unused URL here.

**Comment by dleehr** [comment]
Actually I'll hold off on changing the Docker hub builds for both of these, since an exome-seq job is in progress that will pull them

**Pull request merged**